python/hsfs/feature_group.py

```diff
@@ -309,24 +311,45 @@ def select_all(
         # Arguments
             include_primary_key: If True, include primary key of the feature group
                 to the feature list. Defaults to True.
+            include_foreign_key: If True, include foreign key of the feature group
+                to the feature list. Defaults to True.
+            include_partition_key: If True, include partition key of the feature group
+                to the feature list. Defaults to True.
             include_event_time: If True, include event time of the feature group
                 to the feature list. Defaults to True.
         # Returns
             `Query`. A query object with all features of the feature group.
         """
-        if include_event_time and include_primary_key:
+        if include_event_time and include_primary_key and include_foreign_key:
             return query.Query(
                 left_feature_group=self,
                 left_features=self._features,
                 feature_store_name=self._feature_store_name,
                 feature_store_id=self._feature_store_id,
             )
         elif include_event_time:
```

Comment:
I think this logic would not work now since we have four possible options. e.g. If the users set `include_event_time` = `False`, `include_primary_key` = `True`, `include_foreign_key` = `True` and `include_partition_key` = `False`. The query would not contain foreign keys.

So maybe a better option would be to use a list and append the required keys to it.

"""
removed_keys = []

if not include_event_time:
    removed_keys += [self.event_time]
if not include_primary_key:
    removed_keys += self.primary_key
if not include_foreign_key:
    removed_keys += self.foreign_key
if not include_partition_key:
    removed_keys += self.partition_key

return self.select_except(removed_keys)
"""